File: Programming with C#/3. C# Object-Oriented Programming/04. Object-Oriented Programming Fundamental Principles – Part I/02. Human/Human.cs
using System;
using System.Linq;

abstract class Human
{
    private string firstName = string.Empty;
    private string lastName = string.Empty;

    public Human(string firstName, string lastName)
    {
        this.FirstName = firstName;
        this.LastName = lastName;
    }

    public string FirstName
    {
        get
        {
            return this.firstName;
        }
        set
        {
            if (string.IsNullOrEmpty(value))
                throw new ArgumentException("First name cannot be null or empty");

            this.firstName = value;
        }
    }

    public string LastName
    {
        get
        {
            return this.lastName;
        }
        set
        {
            if (string.IsNullOrEmpty(value))
                throw new ArgumentException("Last name cannot be null or empty");

            this.lastName = value;
        }
    }
}

File: Web Design & Development/5. JavaScript OOP - ✘/03. Advanced OOP/01. School - Prototypal OOP/scripts/Classes/Human.js
define(function(require) {

    var Human = {
        init: function(firstName, lastName, age) {
            this.firstName = firstName;
            this.lastName = lastName;
            this.age = age || 0;

            return this;
        },
        introduce: function() {
            var result = "Name: " + this.firstName + " " + this.lastName + ", " +
                          "Age: " + this.age;

            return result;
        }
    };

    return Human;
})